Adding country support for Germany, retry in case of connection error, option for Janssen

edwillys opened the pull request:
Please feel free to comment on the changes and let me know if you would be willing to merge all or a part of the changes.

edwillys committed:
Adding country support. Now France (fr) and Germany are available (de)
Adding Janssen vaccine option
Adjusting test cases and readme
Retry after 5s in case of failure in connection
In case on many vaccine types, try all of the motives instead of only one.

edwillys committed:
Better handling of country argument
Leaving out deprecated example image on Readme

rbignon commented:
Thank you for this nice contribution. Why changing 'username', 'password' and 'city' to required options?

I would prefer keeping positional arguments, and add --country as an optional parameter (with the default value guessed from the system locales).

edwillys commented:
No worries. `password` is not a mandatory argument. I changed from positional to tagged to keep the consistency with the other options, as I found a bit strange to mix both styles. If you would like to keep it it mixed, I can revert that part.

As for --country, Are not sure about setting the default value to the system locale? Wouldn't this cause confusion in case the locale doesn't match the country the user wants to make an appointment? Also a possible tricky use case would be if a city name is available in multiple countries (can't think of one right now though)

rbignon commented on README.md:
```diff
@@ -1,13 +1,9 @@
 
-<p align="center">
-  <img src="https://raw.githubusercontent.com/rbignon/doctoshotgun/da5f65a1e2ecc7b543376b1549c62004a454b90d/example.svg">
-</p>
```
You do not like this beautiful video? :)

rbignon commented:
> No worries. `password` is not a mandatory argument. I changed from positional to tagged to keep the consistency with the other options, as I found a bit strange to mix both styles. If you would like to keep it it mixed, I can revert that part.

`city` and `email` are mandatory parameters, so it is better to be positional arguments (more user friendly imo).

> As for --country, Are not sure about setting the default value to the system locale? Wouldn't this cause confusion in case the locale doesn't match the country the user wants to make an appointment? Also a possible tricky use case would be if a city name is available in multiple countries (can't think of one right now though)

If you think it is better to be explicit, I prefer to make 'country' the first positional argument.

edwillys committed:
city, username and password are now positional arguments again
Adding country as first positional argument
Aligning readme

edwillys commented:
Done. Only thing is, I didn't update the example GIF from the readme. As it wasn't up to date anymore I deleted it for now. I guess you might want to update it with a snippet from your computer (?)

edwillys committed:
Not looking anymore for Janssen's 2nd shot, as there isn't any
Dealing with different slot response formats
Missing dependencies to make playsound play sound

edwillys committed:
Merge branch 'master' into master

edwillys committed:
Removing playsound dependencies as they are now optional on the main repo

edwillys committed:
Making terminal colors work for windows
Updating readme with python3 mentioned explicitely.

rbignon commented:
Hello,

I've rebased your branch on master and pushed on https://github.com/rbignon/doctoshotgun/tree/de, can you check it's ok for you?

edwillys commented:
Hello, thanks for that. Yeah, it looks fine. Just one comment:
why not keep the ``` self.open( self.BASEURL + '/sessions/new') ``` 
```
    def do_login(self):
        self.open('https://www.doctolib.fr/sessions/new')
```

In the end the login works regardless of the level domain though.

rbignon commented:
> Hello, thanks for that. Yeah, it looks fine. Just one comment:
> why not keep the `self.open( self.BASEURL + '/sessions/new')`
> 
> ```
>     def do_login(self):
>         self.open('https://www.doctolib.fr/sessions/new')
> ```
> 
> In the end the login works regardless of the level domain though.

It's an error during the conflict resolution, I fix it.

rbignon commented:
Merged in master, thank you!
